### ISSUE TITLE
Does not understand alternative META6 dependency format (i.e. Inline::Perl5)

### ISSUE THREAD

**Issue by ugexe** [chloekek/raku-nix#9]
`Inline::Perl5` (and others like `Inline::Python`) use a hash for the e.g. `depends` instead of an array: https://github.com/niner/Inline-Perl5/blob/b6e4e5922c6ff1d734769ce55cc7be602f36ef18/META6.json#L9-L25

See https://design.raku.org/S22.html#depends but note support for this even in zef is flimsy ( see https://github.com/ugexe/zef/blob/e77560688e6a5508b476bd9dba013523aa695b15/lib/Zef/Distribution.pm6#L71-L77 ). Personally I would suggest others to just use the array format ( an array of hash is OK -- see https://deathbyperl6.com/perl-toolchain-summit-2018-meta6-dependency-hello/ ) but for now this is the reality.

**Comment by zopsicle** [comment]
Should be easy to fix. Something that also doesn’t work currently is `:ver<...>`, `:auth<...>`, `:api<...>` etc, they just get discarded. :stuck_out_tongue:

**Comment by ugexe** [comment]
I suspect `"depends" : [ { "name" : "Foo" } ]` also does not work (this is *not* the same as the hash form mentioned earlier) although I have not verified.

Additionally `curl:from<bin>` would search PATH for `curl` (not search for a curl module) and `curl:from<native>` would look for a e.g. `libcurl.so`

**Comment by zopsicle** [comment]
Oh, so there _is_ already syntax for specifying native dependencies! That’s neat, for #1. Although I would still need to keep a mapping from library names to Nix package names (e.g. ssl → openssl), but that is fine.

**Comment by ugexe** [comment]
Yes see "ADD BASIC :FROM<BIN> AND :FROM<NATIVE> SUPPORT" from https://deathbyperl6.com/perl-toolchain-summit-2018-meta6-dependency-hello/
